woodwork/statistics_utils.py

```diff
@@ -148,6 +148,9 @@ def _get_describe_dict(dataframe, include=None, callback=None,
 
 def _get_mode(series):
     """Get the mode value for a series"""
+    if str(series.dtype) == 'boolean':
+        # TODO: remove when pandas fixes bug where mode can't be calculated on BooleanDtype with nans
+        series = series.astype('object')
```

Comment:
A [backport PR](https://github.com/pandas-dev/pandas/pull/42131) has been merged in, so, assuming that means that this change will get into pandas 1.3.0, we should be able to remove this. (tested those changes locally as well to confirm they fix the problems in `_get_mode`)

Reply:
Do you think we should wait to merge in this PR until they release 1.3.0 so we can confirm this was fixed?

Reply:
Possibly. It seems that they're still going to be making various fixes in the mean time, so it might be worth waiting just to confirm that a new bug isn't introduced

Reply:
now that pandas 1.3.0 is in, removed these lines, and the tests still pass :)